File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.swift
class Solution {
    func countKDifference(_ nums: [Int], _ k: Int) -> Int {
        var hashSet = [Int : Int]()
        var ans = 0;
        for item in nums {
            ans += (hashSet[item + k] ?? 0)
            ans += (hashSet[item - k] ?? 0)
            hashSet[item] = (hashSet[item] ?? 0) + 1
        }
        return ans
    }
}

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.cpp
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        unordered_map <int, int> Mp;
        int res = 0;
        for(auto &x: nums) {
            res += Mp[x + k] + Mp[x - k];
            Mp[x] += 1;
        }
        return res;
    }
};